forward/forwarder_test.go

```diff
@@ -112,14 +133,88 @@ func (s *ForwarderTestSuite) TestForward() {
 	s.Equal("Hello, world!", pong.Message)
 }
 
+func (s *ForwarderTestSuite) TestForwardJSONErrorResponse() {
+	var ping Ping
+
+	dest, err := s.sender.Lookup("other 1")
+	s.NoError(err)
+
+	_, err = s.forwarder.ForwardRequest(ping.Bytes(), dest, "test", "/error", []string{"other 1"},
+		tchannel.JSON, nil)
+	s.EqualError(err, "remote error")
+}
+
+func (s *ForwarderTestSuite) TestForwardJSONInvalidEndpoint() {
+	var ping Ping
+
+	dest, err := s.sender.Lookup("other 1")
+	s.NoError(err)
+
+	_, err = s.forwarder.ForwardRequest(ping.Bytes(), dest, "test", "/invalid", []string{"other 1"},
+		tchannel.JSON, &Options{
+			MaxRetries: 1,
+			RetrySchedule: []time.Duration{
+				100 * time.Millisecond,
+			},
+		})
+	s.EqualError(err, "max retries exceeded")
+}
+
+func (s *ForwarderTestSuite) TestForwardThrift() {
+	dest, err := s.sender.Lookup("other 1")
+	s.NoError(err)
+
+	request := &pingpong.PingPongPingArgs{
+		Request: &pingpong.Ping{
+			Key: "success",
+		},
+	}
+
+	bytes, err := SerializeThrift(request)
+	s.NoError(err, "expected ping to be serialized")
+
+	res, err := s.forwarder.ForwardRequest(bytes, dest, "test", "PingPong::Ping", []string{"other 1"},
+		tchannel.Thrift, nil)
+	s.NoError(err, "expected request to be forwarded")
+
+	var response pingpong.PingPongPingResult
+
+	err = DeserializeThrift(res, &response)
+
+	s.Equal("127.0.0.1:3002", response.Success.Source)
```

Comment:
One reason you might do this is because the output of a failed test to `NoError` is less descriptive than to `Equal`.

`NoError` doesn't tell you the name of the thing you tried to check the error against. But `Equal` will output `"Foo" does not equal "Bar"` so you can immediately see in the output what it is failing on.